submissions/Zakkarat/OOP-exercise/index.js

```diff
@@ -8,33 +8,54 @@
 // ======== OBJECTS DEFINITIONS ========
 // Define your objects here
 
-const INHABITANT = {
-    species: {
-        human: 'human'
-    },
-    genders: {
-        male: 'male',
-        female: 'female'
-    }
-};
-const Character = function (name, gender, legs, saying) {
+class Character{
+  constructor(name, gender, saying) {
   this.name = name;
   this.gender = gender;
-  this.legs = legs;
   this.saying = saying;
   }
-  const Cat = function (name, gender, legs, saying) {
-    Character.call(this, name, gender, legs, saying);
-    this.species = "cat";
+}
+class Animal extends Character {
+  constructor(name, gender, saying){
+    super(name, gender, saying);
+    this.legs = 4;
+  }
+}
+class Cat extends Animal {
+  constructor(name, gender, saying){
+    super(name, gender, saying);
+    this.specie = "cat";
+  }
+}
+class Dog extends Animal {
+  constructor(name, gender, saying){
+    super(name, gender, saying);
+    this.specie = "dog";
   }
-  const Dog = function (name, gender, legs, saying) {
-    Character.call(this, name, gender, legs, saying);
-    this.species = "dog";
+}
+class Humanoid extends Character {
+  constructor(name, gender, saying){
+    super(name, gender, saying);
+    this.hands = 2;
+    this.legs = 2;
   }
+}
+class Human extends Humanoid {
+  constructor(name, gender, saying) {
+    super(name, gender, saying);
+    this.specie = "human";
+  }
+}
+class Rat extends Humanoid {
+  constructor(name, gender, saying) {
+    super(name, gender, saying);
+    this.specie = "rat"
+  }
+}
   Character.prototype.sentence = function() {
     let sentence = [
         `Hey there, my name is <strong>${this.name}</strong> `,
-        `and I am a ${this.species}! `,
+        `and I am a ${this.specie}! `,
         `My gender is ${this.gender}, `,
         `in addition I have ${this.legs} legs`,
         ` and ${this.hands} hands.`,
```

Comment:
I'm not sure that is good idea to mess prototype style with es6 classes. I think it would be better, put you sentence to class character in constructor or method of class.